gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java

```diff
@@ -393,7 +393,10 @@ private void configureServices(){
     registerServicesInLauncher();
 
     // Register Scheduler to listen to changes in Flows
-    if (configuration.isSchedulerEnabled()) {
+    // In warm standby mode, instead of scheduler we will add orchestrator as listener
```

Comment:
Should we separate the compiler with the orchestrator? What we really want here is just the compiler to be used, and not any of the orchestration capabilities.

Reply:
So I think the only uncertainty here would be if a user has a scheduled flow, should we store their changes permanently if their flow cannot be run immediately due to quota?
Currently, the behavior is that **we reject all changes if the flow is set to run immediately and quota is overrun**
If the flow is not set to run immediately, it should pass.

In other systems like ADF, the schedule of a flow and the configurations of a flow are separate. So it can be argued that we accept changes, but refuse to schedule them if their quota is overrun.

Reply:
I see we seems to have different compiler class based on the config, but on flow catalog side we need the class name to get the compile result, that's why I use orchestrator as it's constant...
Another thing is I see is we need to set the compiler as active and listen to topology catalog's change, so want to make sure we only have one compiler in the service